libs/market-depth/src/lib/depth-chart.tsx

```diff
@@ -161,7 +162,15 @@ export const DepthChartContainer = ({ marketId }: DepthChartManagerProps) => {
 
   return (
     <AsyncRenderer loading={loading} error={error} data={data}>
-      {depthData && <DepthChart {...depthData} theme={theme} />}
+      {depthData && (
+        <DepthChart
+          {...depthData}
+          theme={theme}
+          volumeFormat={(volume) =>
```

Comment:
We should also pass a value for `priceFormat`. The chart defaults to 2dp if we don't provide a value.

Reply:
hmm, there are a lot of `formatLevel` function calls going on a few lines above, which seemingly pre-format the prices going in the chart data. are those redundant? not sure i fully understand their purpose

Reply:
They seem badly named. I'd probably call them `parseLevel` or similar. It looks like they take the raw API data which are strings and converts it to numbers.

Reply:
okay, removed the decimal formatting from the formatLevel function and renamed it to parseLevel

Reply:
We are surrounded by strangely named functions!

I think we still need `addDecimal` because it converts the string (integer) we get from the API to a string (decimal floating point style, e.g. NNNN.MMMMM). We can then convert this to a number to pass to the chart.

What we could do with is a function that takes the string from the API and returns the corresponding JS number. We might already have one but I can't see it. There is `toBigNum` but the charts don't support it.

(Off topic for this PR but presumably we work with numbers the way we do because we are concerned about precision and accuracy. This should impact the charts too. I'll put some thought into it) 

Now if volume is encoded in the same way we also need to convert that from an integer string to a decimal string.

All of the above is based on my knowledge of the API data schema from last year. It's possible we have changed how we encode numbers in a way which materially affects the charts but I've not seen anything to suggest that.

Reply:
okay, i think i understand what the problem was, namely that the decimalPlaces and the positionDecimalPlaces are different, so to be accurate along the x and y axes we need to have both numeric values using the correct decimals for each. fixed this. not sure what else we're missing in this PR.